buildSrc/src/main/groovy/docker-import-backup-lxc-config.gradle

```diff
@@ -29,6 +33,70 @@ copyLiferayLXCRepositoryConfiguration = {
 	return true
 }
 
+Closure<Boolean> updateGradlePropertiesWithLiferayVersion = {
+	String environmentName ->
+
+	File environmentDescriptorFile = new File(new File(projectDir, config.liferayLXCRepositoryPath), "automation/environment-descriptors/${environmentName}.json")
+
+	if (!environmentDescriptorFile.exists()) {
+		return false
+	}
+
+	JsonSlurper jsonSlurper = new JsonSlurper()
+
+	Object environmentDescriptor = jsonSlurper.parse(environmentDescriptorFile)
+
+	String liferayImage = environmentDescriptor["liferay-image"]
+	String liferayHotfix = environmentDescriptor["hotfix"]
+
+	Pattern releasePattern = ~'[0-9]+\\.q[1-4]\\.[0-9]+'
+
+	Matcher releaseMatcher = releasePattern.matcher(liferayImage)
+
+	if (!releaseMatcher.find()) {
+		println "did not match ${liferayImage}"
+		return false
+	}
+
+	String release = releaseMatcher.group()
+
+	if ((release.indexOf(".q1") != -1) && !release.startsWith("2024.")) {
+		release = release + "-lts"
+	}
+
+	String workspaceProduct = "dxp-${release}"
+	String liferayHotfixURL = "https://releases-cdn.liferay.com/dxp/hotfix/${release}/liferay-dxp-${release}-${liferayHotfix}.zip"
+
+	File gradlePropertiesFile = new File(projectDir, "gradle.properties")
+
+	List<String> gradlePropertiesLines = []
+	
+	gradlePropertiesFile.eachLine {
```

Comment:
Just a stylistic note, but you can use `collect` to transform an existing array (like `Array.map` in Javascript).

Reply:
Updated in c9a2b018e442fbdcdf0af3ed6d305b23de309c2a